v2: add `Send` bound to most of the API

steveeJ opened the pull request:
This is required for consumers who want to use these methods in a
multi-threaded program.

Along the way some error conversion had to be changed.

lucab commented on src/v2/manifest/mod.rs:
```diff
@@ -162,15 +162,18 @@ impl Client {
                 trace!("HEAD {:?}", url);
             })
-            .and_then(|r| {
+            .and_then(|r| -> std::result::Result<_, ::errors::Error> {
```
Isnt't this return type the same as `::errors::Result<_>`?
(I'm not sure why this needs additional hinting, but I trust that you fought the compiler before ending up with this)

steveeJ replied:
This was part of debugging and leftover. Thanks for catching!

steveeJ commented:
@lucab I addressed your comment, and also added a commit on top to address all compiler warnings. PTAL and let me know if you'd prefer a separate PR for that or it is fine here.
